### PR TITLE
make "tags" to be array like "tags" in grok filter

### PR DESCRIPTION
**Pull request by garyelephant**
This pull request will make json filter to be the same as "tags" in grok filter, which fix this situation:

```
{
     "tags" => "_jsonparsefailure_grokparsefailure",
     "@version" => "1",
     "@timestamp" => "2015-04-13T09:06:48.887Z",
     ...
}
```

if _jsonparsefailure and _grokparsefailure occur in the same time , the correct output will be:

```
{
     "tags" => [
         [0] "_jsonparsefailure",
         [1] "_grokparsefailure"
     ],
     "@version" => "1",
     "@timestamp" => "2015-04-13T09:06:48.887Z",
     ...
}
```

### CODE REVIEW

**Comment by suyograo** [comment]
@garyelephant thanks! could you add tests checking for this behaviour please..let us know if you need help writing tests

**Comment by suyograo** [comment]
@garyelephant also, can you please perform step 2 of https://github.com/elasticsearch/logstash/blob/master/CONTRIBUTING.md#contribution-steps

**Comment by suyograo** [comment]
@garyelephant can you add similar tests here too? Like you added for https://github.com/logstash-plugins/logstash-filter-json/pull/6

File: spec/codecs/json_spec.rb [hunk 1]
```diff
@@ -64,6 +64,48 @@
         insist { decoded } == true
       end
     end
+
+    context "when json could not be parsed" do
+
+      let(:message)    { "random_message" }
+
+      it "add the failure tag" do
+        subject.decode(LogStash::Json.dump(message)) do |event|
```
Comment (suyograo) [inline spec/codecs/json_spec.rb]: no need to do `LogStash::Json.dump(message)`. This is done internally in the codec

File: spec/codecs/json_spec.rb [hunk 2]
```diff
@@ -64,6 +64,48 @@
...
+          expect(event['tags']).to include "_jsonparsefailure"
+        end
+      end
+
+      context "there are two different errors added" do
+
+        let(:json)     { '{"message":"random_message", "tags":["_anotherkinfoffailure"]}' }
+
+        it "pile the different error messages" do
+          subject.decode(LogStash::Json.dump(json)) do |event|
```
Comment (suyograo) [inline spec/codecs/json_spec.rb]: same here. No need to `LogStash::Json.dump(message)`

**Comment by suyograo** [comment]
@garyelephant there are some issues with the tests, as I have commented. Let me know if you need more info/help. Thanks for the patience!

File: lib/logstash/codecs/json.rb [hunk 3]
```diff
@@ -35,7 +35,7 @@ def decode(data)
       yield LogStash::Event.new(LogStash::Json.load(data))
     rescue LogStash::Json::ParserError => e
       @logger.info("JSON parse failure. Falling back to plain-text", :error => e, :data => data)
-      yield LogStash::Event.new("message" => data, "tags" => "_jsonparsefailure")
+      yield LogStash::Event.new("message" => data, "tags" => ["_jsonparsefailure"])
```
Comment (jordansissel) [inline lib/logstash/codecs/json.rb:38]: Oops, yeah I agree this is a bug and this patch fixes it.

**Comment by suyograo** [comment]
@garyelephant need any help on these?

**Comment by garyelephant** [comment]
@suyograo fixed the issues

**Comment by suyograo** [comment]
LGTM

**Comment by suyograo** [comment]
Merged https://github.com/logstash-plugins/logstash-codec-json/commit/0eb5dcc612255edaa90b3b40596f7abe6ad9c188

**Pull request closed without merging**